test/functional/legacy/assert_spec.lua

```diff
@@ -0,0 +1,159 @@
+local ffi = require('ffi')
+local helpers = require('test.functional.helpers')
+local nvim, call = helpers.meths, helpers.call
+local clear, eq = helpers.clear, helpers.eq
+local write_file, execute = helpers.write_file, helpers.execute
+
+local function expected_errors(errors)
+  eq(errors, nvim.get_vvar('errors'))
+end
+
+local function expected_empty()
+  eq({}, nvim.get_vvar('errors'))
+end
+
+local function source(code)
+  local tmpname = os.tmpname()
+
+  if ffi.os == 'OSX' and string.match(tmpname, '^/tmp') then
+   tmpname = '/private'..tmpname
+  end
+
+  write_file(tmpname, code)
+  nvim.command('source '..tmpname)
+  os.remove(tmpname)
+  return tmpname
```

Comment:
I thought this way is too ad-hoc to use for any functional test, but ok, I'll do that.